tracer/src/Datadog.Trace.SourceGenerators/AspectsDefinitions/AspectsDefinitionsGenerator.cs

```diff
@@ -221,46 +257,48 @@ private static string GetFullName(ITypeSymbol? type, bool extractOnlyBaseType =
         return name;
     }
 
-    private static string GetFullName(ISymbol? member)
+    private static string GetFullName(IMethodSymbol method)
     {
-        if (member != null && member is IMethodSymbol methodSymbol)
+        string arguments = string.Join(",", method.Parameters.Select(a => GetFullName(a.Type, false)));
+        return $"{method.Name}({arguments})";
+    }
+
+    private static string GetArgument(TypedConstant customAttributeArgument)
+    {
+        if (customAttributeArgument.Type is null) { return NullLiteral; }
+        else if (customAttributeArgument.Kind == TypedConstantKind.Primitive)
+        {
+            var value = customAttributeArgument.Value?.ToString() ?? NullLiteral;
+            if (customAttributeArgument.Type!.Name == "String") { return $"\"{value}\""; }
+            return value;
+        }
+        else if (customAttributeArgument.Kind == TypedConstantKind.Enum)
         {
-            string arguments = string.Join(",", methodSymbol.Parameters.Select(a => GetFullName(a.Type, false)));
-            return $"{member.Name}({arguments})";
+            if (customAttributeArgument.Type!.Name == "AspectFilter") { return ((AspectFilter)customAttributeArgument.Value!).ToString(); }
+            else if (customAttributeArgument.Type!.Name == "AspectType") { return ((AspectType)customAttributeArgument.Value!).ToString(); }
+            else if (customAttributeArgument.Type!.Name == "VulnerabilityType") { return ((VulnerabilityType)customAttributeArgument.Value!).ToString(); }
+            var type = Resolve(customAttributeArgument);
+            return Enum.ToObject(type, customAttributeArgument.Value).ToString();
+        }
+        else if (customAttributeArgument.Kind == TypedConstantKind.Array)
+        {
+            var elementType = Resolve(customAttributeArgument);
+            var values = customAttributeArgument.Values;
+            return $"[{string.Join(",", values.Select(GetArgument))}]";
         }
 
         return string.Empty;
     }
 
-    private static string GetSource(IReadOnlyList<string> definitions)
+    internal struct ClassAspects
```

Comment:
IIRC structs are compared by value, but the comparison uses relatively slow reflection - making it a `record struct` is more efficient I believe

```suggestion
    internal readonly record struct ClassAspects
```

Reply:
Yo are [right](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record)